File: Java/Hard/First-Missing-Positive.java
class Solution {
  public int firstMissingPositive(int[] nums) {
    HashSet<Integer> setNums = new HashSet<>();
    for (int i = 0; i < nums.length; i++) {
      setNums.add(nums[i]);
    }
    int i = 1;
    while (i <= nums.length) {
      if (setNums.contains(i)) {
        i++;
      } else {
        return i;
      }
    }
    return nums.length + 1;
  }
}

File: Python/Hard/First-Missing-Positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        setNums = {set(nums)}
        i = 1
        while i<=len(nums):
            if i in setNums:
                i+=1
            else:
                return i
        return len(nums) + 1
